### ISSUE TITLE
活用できそうなLoss関数

### ISSUE THREAD

**Issue by habakan** [katsu1110/NNNS#4]
3/16時点でのモブプロ会の議論も含めてLoss関数の一覧をまとめておきました。  
Loss関数に対する議論や、これ以外のLoss関数について書いていただけると知見がたまりますのでよろしくおねがいします！

### Regression Loss
- [keras regression losses](https://keras.io/api/losses/regression_losses/)
- HuberやLogCoshなどはすぐに検証できるかも？
### ピアソン or スピアマン相関系
- Metricsの一つであるCorrを直接最適化することを試みるloss
- 参加者のなかで、うまくいかなったという報告あり
### BCE
- 3の場合などは、0~3までを1として扱い各次元で2値分類を行う？
- kazutsugiの時代はうまくいっていた（targetの分布が一様だったため）
- ラベル不均衡が問題なら、[Focal Loss](https://paperswithcode.com/method/focal-loss)のような不均衡に対応できるLossを利用するなど？
### Weighted Kappa Loss
- 順序性のある離散的な値に対して評価する関数
- [tf document](https://www.tensorflow.org/addons/api_docs/python/tfa/losses/WeightedKappaLoss#:~:text=Weighted%20Kappa%20loss%20was%20introduced%20in%20the%20Weighted,in%20[%20%E2%88%92%20%E2%88%9E,%20log%202],%20where%20log)
### Label Distribution Learning
- [解説記事](https://tech-blog.optim.co.jp/entry/2021/03/30/100000#label-distribution-learning)
- [論文](https://arxiv.org/abs/1611.01731)
- 順序性のあるラベルに対して、one-hotでcross entropyを計算するのではなく、正解ラベルをμとした正規分布に変換してcross entropyをかける
- やっていることは、順序性を考慮したlabel smoothing？
- クラス数の少ないnumerai targetにどのように対応するかどうか？
### その他にでてきたidea
- ランキング学習の枠組みで学習できないか？
- FNCを向上させるlossを定義できないか？

**Comment by katsu1110** [comment]
@habakan ありがとうございます！

BCE lossに関連してですが、見込みがあればlabel smoothingを上手く使えないか検討してもいいかなと思っています。Focal lossと組み合わせることもできそうなので、個人的には期待しています。

https://www.kaggle.com/code/jimitshah777/bilinear-efficientnet-focal-loss-label-smoothing/notebook

**Comment by habakan** [comment]
@katsu1110 
わりと、BCEでもともとうまくいっていたということは大きい気がしていて、そこのボトルネックを解消すると何か生まれそうな気がしますね。  
個人的に複数次元出力はやったことがなかったので気になってます！
